drivers/hmis/app/graphql/mutations/submit_form.rb

```diff
@@ -78,13 +78,19 @@ def resolve(input:)
         # Perform any side effects
         perform_side_effects(record)
 
-        if record.is_a? Hmis::Hud::HmisService
+        case record
+        when Hmis::Hud::HmisService
           record.owner.save! # Save the actual service record
           record = Hmis::Hud::HmisService.find_by(owner: record.owner) # Refresh from View
-        elsif record.is_a? HmisExternalApis::AcHmis::ReferralRequest
+        when HmisExternalApis::AcHmis::ReferralRequest
           HmisExternalApis::AcHmis::CreateReferralRequestJob.perform_now(record)
-        elsif record.is_a?(Hmis::Hud::Enrollment) && record.new_record?
-          record.save_in_progress
+        when Hmis::Hud::Enrollment
+          record.client.save! if record.client.changed? # Enrollment form may create or update client
```

Comment:
This could be accomplished more cleanly with an `autosave: true` on Enrollment's relationship to Client, but I wasn't sure if that is desirable or not.   (Likewise we could switch to `validates: true` regardless of context).

@eanders is there a particular reason why we don't autosave by default on the hud assoc relationships?

Reply:
Yeah, the warehouse does everything in batches, and rarely changes data since it's coming from somewhere else.  autosave can lead to massive N+1, so we avoid it.  You could probably put it on the HMIS version of the models if you want though.  Not sure how much churn that'll make.

Reply:
Ok gotcha, I figured it was a performance concern. Part of me feels like it does make sense to do it on the HMIS models, since the application is all about data entry. I'm not sure how often this will happen of non-assessment forms needing to update related records. Maybe I'll come back to that if I see it becomes a larger pattern.